classes/class-p4-campaign-exporter.php

```diff
@@ -29,9 +29,9 @@ public function single_post_export_data() {
 			if ( ! ( isset( $_GET['post'] ) || isset( $_POST['post'] ) || ( isset( $_REQUEST['action'] ) ) ) ) {
 				wp_die( 'No post to export has been supplied!' );
 			}
-			$post_id = ( isset( $_GET['post'] ) ? $_GET['post'] : $_POST['post'] );
+			$post_id = ( isset( $_GET['post'] ) ? filter_input( INPUT_GET, 'post', FILTER_SANITIZE_NUMBER_INT ) : filter_input( INPUT_POST, 'post', FILTER_SANITIZE_NUMBER_INT ) );
```

Comment:
No need to check if 'post' key isset since we are going to use filter funciton to access it.
Is it coming from a GET or a POST request? It may come from both? Then we can use INPUT_REQUEST to make sure we get it right. And we can even remove the first `if` statement.
`$post_id = filter_input( INPUT_REQUEST, 'post', FILTER_SANITIZE_NUMBER_INT );`
```
if ( ! empty( $post_id ) && isset( $_REQUEST['action'] ) ) {
	include get_template_directory() . '/exporter.php';
} else {
	wp_die( 'No post to export has been supplied!' );
}
```

Reply:
I was also looking for `INPUT_REQUEST` type. but I didn't found it in available type options in documentation.(https://www.php.net/manual/en/function.filter-input.php).

Reply:
Oh ok, I saw it is defined (in php7.3 at least) and thought we could use it.
But still no need to check if isset since the filter_input function does that anyway.
We could also validate the 'action' key if we know what value it must have.